backend-rust/src/graphql_api.rs

```diff
@@ -2810,14 +2810,99 @@ impl Account {
 
     async fn transactions(
         &self,
-        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: i32,
+        ctx: &Context<'_>,
+        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: Option<u64>,
         #[graphql(desc = "Returns the elements in the list that come after the specified cursor.")]
-        after: String,
-        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: i32,
+        after: Option<String>,
+        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: Option<u64>,
         #[graphql(desc = "Returns the elements in the list that come before the specified cursor.")]
-        before: String,
+        before: Option<String>,
     ) -> ApiResult<connection::Connection<String, AccountTransactionRelation>> {
-        todo_api!()
+        let config = get_config(ctx)?;
+        let pool = get_pool(ctx)?;
+        let query = ConnectionQuery::<i64>::new(
+            first,
+            after,
+            last,
+            before,
+            config.contract_connection_limit,
+        )?;
+
+        let mut txs = sqlx::query_as!(
+            Transaction,
+            r#"SELECT * FROM (
+                SELECT
+                    index,
+                    block_height,
+                    block_index,
+                    hash,
+                    ccd_cost,
+                    energy_cost,
+                    sender,
+                    type as "tx_type: DbTransactionType",
+                    type_account as "type_account: AccountTransactionType",
+                    type_credential_deployment as "type_credential_deployment: CredentialDeploymentTransactionType",
+                    type_update as "type_update: UpdateTransactionType",
+                    success,
+                    events as "events: sqlx::types::Json<Vec<Event>>",
+                    reject as "reject: sqlx::types::Json<TransactionRejectReason>"
+                FROM transactions
+                WHERE
+                    sender = $1
```

Comment:
Is this correct? I.e. are transactions associated with the sender account? Or do we need to have a receiver column and use that?

Reply:
The current CCDScan seems to return all kinds of transactions that the account is somehow involved in e.g. even if the account receives CCD (and is not the initiator/sender of the transaction). I even wonder if ContractUpdate transactions appear if the account gets CCD paid out to its address from a contract address (haven't seen that event yet, so not sure).

Transactions such as where the account receives CCD (not sends the CCD).
- Simple transfer 
- Simple CCD transfer with memo
- Simple CCD transfer with a schedule